### PR TITLE
Add ability to provide preset response options in `gr.Chatbot` / `gr.ChatInterface`

### PR DESCRIPTION
**Pull request by abidlabs**
Adds an optional `options` field to `MessageDict` that can take a list of dictionaries of `value` and optionally, a separate `label`, which produces preset options that a user can click on. 

To see it in action, please run: `demo/chatinterface_options/run.py`

https://github.com/user-attachments/assets/ed9b7672-1b5d-41b2-8479-7a5c9326ca56

Closes: https://github.com/gradio-app/gradio/issues/8698
Closes: https://github.com/gradio-app/gradio/issues/4763
Closes: https://github.com/gradio-app/gradio/issues/6372

I'm working on improving the ChatInterface docs so hold off on reviewing those, but once the docs are ready, it'll also close:

Closes: https://github.com/gradio-app/gradio/issues/9917

### CODE REVIEW

**Comment by abidlabs** [comment]
Docs are ready too now!

**Review by hysts — APPROVED** [review]
LGTM!

**Comment by abidlabs** [comment]
Tweaked the styling a bit (cc @hannahblair):


https://github.com/user-attachments/assets/2979c3ef-8881-44fb-bdaf-9c0c9ddf64f8

**Comment by abidlabs** [comment]
Tweaked the styling a bit (cc @hannahblair):


https://github.com/user-attachments/assets/428c47d5-0313-420c-a811-520f75b63b4b

**Comment by abidlabs** [comment]
Maybe `presets` or `responses` would be a better name instead of `options`? Open to suggestions on the api

File: js/chatbot/shared/ChatBot.svelte [hunk 1]
```diff
@@ -459,11 +485,6 @@
 		justify-content: center;
 	}
 
-	.example:hover {
```
Comment (dawoodkhan82) [inline js/chatbot/shared/ChatBot.svelte:462]: why did we remove this?

**Review by dawoodkhan82 — APPROVED** [review]
Left one small comment. Otherwise lgtm!

File: js/chatbot/shared/ChatBot.svelte (hunk 1 above)
Comment (abidlabs) [reply js/chatbot/shared/ChatBot.svelte:462]: Accident! Will restore

**Comment by abidlabs** [comment]
Thanks @hysts and @dawoodkhan82 for reviewing!

**Pull request merged**